head.go

```diff
@@ -480,10 +486,10 @@ func (h *Head) Init(minValidTime int64) error {
 		return nil
 	}
 	level.Warn(h.logger).Log("msg", "encountered WAL error, attempting repair", "err", err)
+	h.metrics.walCorruptionsTotal.Inc()
```

Comment:
I believe detect WAL corruptions when we call `loadWAL`. So do we also need to increment it for this: https://github.com/prometheus/tsdb/blob/9e51d56e08958f22f55daf26795ee477def7797e/head.go#L471-L473

And also maybe a small test for that?

Reply:
When this happens Prometheus will exist, so what is the point of an increment there?

Reply:
If there is no way to recover this `Inc()` info after we return, then there is no need of adding it here.

Reply:
Maybe add the metrics directly to `wal.Repair()` so we can know when there is a corruption and whether or not it has been repaired?